justfile

```
@@ -0,0 +1,234 @@
+    "rnote-compose=" + log_level
+#export G_MESSAGES_DEBUG := "all"
+
+default:
+    just --list
+
+prerequisites:
+    #!/usr/bin/env bash
+    set -euxo pipefail
+
+    if [[ ('{{linux_distr}}' =~ 'fedora') || ('{{linux_distr}}' =~ 'rhel') || ('{{linux_distr}}' =~ 'alma') ]]; then
+        {{sudo_cmd}} dnf install -y \
+            gcc gcc-c++ clang clang-devel python3 make cmake meson just git appstream gettext desktop-file-utils \
+            shared-mime-info kernel-devel gtk4-devel libadwaita-devel poppler-glib-devel poppler-data alsa-lib-devel \
+            appstream-devel
+    elif [[ '{{linux_distr}}' =~ 'debian' || '{{linux_distr}}' =~ 'ubuntu' ]]; then
+        {{sudo_cmd}} apt-get update
+        {{sudo_cmd}} apt-get install -y \
+            build-essential clang libclang-dev python3 make cmake meson just git appstream gettext desktop-file-utils \
+            shared-mime-info libgtk-4-dev libadwaita-1-dev libpoppler-glib-dev libasound2-dev libappstream-dev
+    else
+        echo "Unable to install system dependencies, unsupported distro."
+        exit 1
+    fi
+
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+    . "$HOME/.cargo/env"
+    curl -L --proto '=https' --tlsv1.2 -sSf \
+        https://raw.githubusercontent.com/cargo-bins/cargo-binstall/main/install-from-binstall-release.sh | bash
+    cargo binstall -y cargo-nextest
+
+prerequisites-flatpak: prerequisites
+    #!/usr/bin/env bash
+    set -euxo pipefail
+
+    if [[ ('{{linux_distr}}' =~ 'fedora') || ('{{linux_distr}}' =~ 'rhel') || ('{{linux_distr}}' =~ 'alma') ]]; then
+        {{sudo_cmd}} dnf install -y \
+            flatpak flatpak-builder
+    elif [[ '{{linux_distr}}' =~ 'debian' || '{{linux_distr}}' =~ 'ubuntu' ]]; then
+        {{sudo_cmd}} apt-get update
+        {{sudo_cmd}} apt-get install -y \
+            flatpak flatpak-builder
+    else
+        echo "Unable to install system dependencies, unsupported distro."
+        exit 1
+    fi
+
+    flatpak remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo
+    flatpak install -y org.gnome.Platform//48 org.gnome.Sdk//48 org.freedesktop.Sdk.Extension.rust-stable//24.08 \
+        org.freedesktop.Sdk.Extension.llvm19//24.08
+
+prerequisites-dev: prerequisites
+    #!/usr/bin/env bash
+    set -euxo pipefail
+
+    if [[ ('{{linux_distr}}' =~ 'fedora') || ('{{linux_distr}}' =~ 'rhel') || ('{{linux_distr}}' =~ 'alma') ]]; then
+        {{sudo_cmd}} dnf install -y \
+            yamllint yq opencc-tools
+    elif [[ '{{linux_distr}}' =~ 'debian' || '{{linux_distr}}' =~ 'ubuntu' ]]; then
+        {{sudo_cmd}} apt-get update
+        {{sudo_cmd}} apt-get install -y \
+            yamllint yq opencc
+    else
+        echo "Unable to install system dependencies, unsupported distro."
+        exit 1
+    fi
+
+    if [[ "{{ci}}" != "true" ]]; then
+        cp -f build-aux/git-hooks/pre-commit.hook .git/hooks/pre-commit
```

We should probably use a symlink here, so that changes are applied without having to re-run this command.

```suggestion
        ln -sf build-aux/git-hooks/pre-commit.hook .git/hooks/pre-commit
```

---

good idea!